### DISCUSSION TITLE
phase-randomized correlation values output

### DISCUSSION THREAD

**Discussion by arfriedman** [LinkedEarth/Pyleoclim_util#622] [Q&A] [answered]
Greetings,

I just started using pyleoclim -- thanks for this excellent package.

I am trying to construct a confidence interval for the correlation statistic (r) using [phased-randomized correlation](https://pyleoclim-util.readthedocs.io/en/latest/utils/introduction.html#pyleoclim.utils.correlation.corr_isospec).
Is there a way to obtain the significant value of r (or the full distribution of r values) from the correlation output? 

Best,
Andrew

**Comment by CommonClimate** [comment]
Hi Andrew, glad you are enjoying the package. As explained [here](http://linked.earth/PyleoTutorials/notebooks/L2_correlations.html#effect-of-persistence), Pyleoclim's `Corr` objects contain :
- r (float), the correlation statistic
- p (float), the p-value, which is used to decide whether the correlation is in any way remarkable (“significant”)
- signif, a boolean indicating whether significance was reached, at the indicated level (0.05 by default)

We do not currently export the distribution of surrogate r values, but I suppose we could. However, doesn't `signif` or `p` give you what you want?

**Reply by arfriedman** [reply]
Thank you for your reply @CommonClimate.

In addition to the `Corr` object contents, I would be interested in using the distribution of surrogate r values to visually represent how anomalous the calculated r value is. For example, see [this paper](https://doi.org/10.1002/2017GL072582) (Figures 1f, 2c-d, and 2h). [I previously made these in Matlab].

If I understand correctly, could I do this using the values `rSim` from `corr_isospec` in **correlation.py**?

**Reply by CommonClimate** [reply, marked as answer]
Yes, rSim is that variable but:
1. it is not currently exported by `corr_isospec`
2. the latest version of the package (1.1 and later) no longer uses this function. Instead, everything is done through the `SurrogateSeries` class and the `correlation.py:association()` function. 

You can either make your own version of `corr_isospec`, which would do the job for this (if all you care about is Pearson's correlation, and not a wider set of metrics) or open an issue to ask us to export the simulated values of r in the correlation object. @khider will likely have objections because this is a change to the API, but since we're adding a field and not renaming/removing, I don't anticipate that impacting backward-compatibility, so it should be OK.

PS: please say *bonjour* to Gilles, Myriam and Guillaume

**Reply by arfriedman** [reply]
Thanks @CommonClimate. I will pursue both these suggestions.
PS. _Merci_, will do!

**Reply by CommonClimate** [reply]
addressed by #634
